### PR TITLE
Skip distributed tests if built with USE_DISTRIBUTED=0

### LINKED ISSUE #70676: Tests fail if built with USE_DISTRIBUTED=0
### 🐛 Describe the bug

Running `python test/run_test.py` fails if Pytorch was compiles with `USE_DISTRIBUTED=0`.

It happens because the distributed tests are launched regardless the availability of `distributed` package.

The expected behaviour is for distributed tests to be skipped in such case.

The example error log is: 

```
Running distributed/_sharded_optim/test_sharded_optim ... [2022-01-05 15:07:05.368855]
Executing ['/home/soul/.pyenv/versions/pytorch/bin/python', 'distributed/_sharded_optim/test_sharded_optim.py'] ... [2022-01-05 15:07:05.368921]
Traceback (most recent call last):
  File "/home/soul/pytorch/test/distributed/_sharded_optim/test_sharded_optim.py", line 5, in <module>
    import torch.distributed._sharded_tensor as sharded_tensor
  File "/home/soul/pytorch/torch/distributed/_sharded_tensor/__init__.py", line 6, in <module>
    from torch.distributed._sharding_spec import (
  File "/home/soul/pytorch/torch/distributed/_sharding_spec/__init__.py", line 1, in <module>
    from .api import (
  File "/home/soul/pytorch/torch/distributed/_sharding_spec/api.py", line 21, in <module>
    class DevicePlacementSpec(PlacementSpec):
  File "/home/soul/pytorch/torch/distributed/_sharding_spec/api.py", line 29, in DevicePlacementSpec
    device: torch.distributed._remote_device
AttributeError: module 'torch.distributed' has no attribute '_remote_device'
Traceback (most recent call last):
  File "/home/soul/pytorch/test/run_test.py", line 1101, in <module>
    main()
  File "/home/soul/pytorch/test/run_test.py", line 1079, in main
    raise RuntimeError(err_message)
RuntimeError: distributed/_sharded_optim/test_sharded_optim failed!
```
although, it's just a first test to fail the testing process.

I reported this problem earlier as #70560, but since then I did some digging and I'm closing the old report due to the lack of details.

### Versions

Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.22.0

Python version: 3.8 (64-bit runtime)
Is CUDA available: N/A
CUDA runtime version: 10.2.89
GPU models and configuration: GPU 0: GeForce RTX 2080 Ti
Nvidia driver version: 460.91.03
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.4
[pip3] pytorch-sphinx-theme==0.0.24
[pip3] torch==1.11.0a0+gitd1dbcb1
[pip3] torchvision==0.11.1
[conda] Could not collect

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

### PR DESCRIPTION
**Pull request by lithuak**
Fixes #70676

### CODE REVIEW

**Comment by pritamdamania87** [comment]
@janeyx99 The changes here LGTM, however I was wondering if someone from the dev-infra team can take a look and stamp this PR?

File: test/run_test.py [hunk 1]
```diff
@@ -969,6 +969,11 @@ def get_selected_tests(options):
             which_shard, num_shards, selected_tests, TEST_TIMES_FILE
         )
 
+    # skip all distributed tests if distributed package is not available.
+    if not dist.is_available():
+        selected_tests = exclude_tests(DISTRIBUTED_TESTS, selected_tests,
+                                       "PyToch is built without distributed support.")
```
Comment (janeyx99) [inline test/run_test.py:975]:
```suggestion
                                       "PyTorch is built without distributed support.")
```
nit

**Review by janeyx99 — APPROVED** [review]

**Comment by lithuak** [comment]
@janeyx99 The test failure looks false negative to me:

```
FAILED test/onnx/test_operators.py::TestOperators::test_c2_op - AssertionErro...

Results (230.33s):
     198 passed
       1 failed
         - test/onnx/test_operators.py:728 TestOperators.test_c2_op
      11 skipped
```

My change shouldn't affect tests in test/onnx/test_operators.py in any way.

Similar failures happened recently for my PR's that we merely changing the docstrings

**Comment by janeyx99** [comment]
Yes, looking at the dashboard here https://hud2.pytorch.org/minihud?name_filter=linux-xenial-py3.7-clang7-onnx%20/%20test%20(default,%201,%202,%20linux.2xlarge) shows that the onnx failure was an upstream failure

**Pull request closed without merging**